services/filemanager/azureBlobStoragemanager.go

```diff
@@ -77,6 +75,24 @@ func (manager *AzureBlobStorageManager) getContainerURL() (azblob.ContainerURL,
 	return containerURL, nil
 }
 
+func (manager *AzureBlobStorageManager) getCredentials() (azblob.Credential, error) {
+	if manager.useSASTokens() {
+		return azblob.NewAnonymousCredential(), nil
+	}
+
+	accountName, accountKey := manager.Config.AccountName, manager.Config.AccountKey
+	if accountName == "" || accountKey == "" {
+		return nil, errors.New("either the AccountName or AccountKey is not correct")
+	}
+
+	// Create a default request pipeline using your storage account name and account key.
+	return azblob.NewSharedKeyCredential(accountName, accountKey)
+}
+
+func (manager *AzureBlobStorageManager) useSASTokens() bool {
+	return manager.Config.UseSASTokens != nil && *manager.Config.UseSASTokens
```

Comment:
We can simplify this check greatly in case we use bool

Reply:
The reason we started using *bool is to handle if the config key is not present. In that case, the pointer will point to nil.
And we can handle that case as well instead of going for default as false.

Reply:
I agree that if this new flag had the opposite default value, things would be different. But in this specific scenario isn't it enough to assume a default value of `false`?

Reply:
Yes @atzoum you are correct. Even if the flag is not present it would be correct to assume that it should be false.

Reply:
Done.